### PR TITLE
Improve `collection` related things that reusing a immutable object instead of creating a new object

### PR DESCRIPTION
**Pull request by asdf2014**
Improve `collection` related things that reusing a immutable object instead of creating a new object
 * `Collections.singletonList(something)` is immutable whereas `Arrays.asList(something)` is a fixed size List representation of an Array where the List and Array gets joined in the heap.
 * `Collections.emptyList()` reuses an object instead of creating a new object as it will be the case with `Arrays.asList()`.

### CODE REVIEW

File: extensions-contrib/parquet-extensions/src/main/java/io/druid/data/input/parquet/ParquetExtensionsModule.java [hunk 1]
```diff
@@ -34,8 +34,8 @@
   @Override
   public List<? extends Module> getJacksonModules()
   {
-    return Arrays.asList(
-        new SimpleModule("ParuqetInputRowParserModule")
+    return Collections.singletonList(
+        new SimpleModule("ParquetInputRowParserModule")
```
Comment (leventov) [inline extensions-contrib/parquet-extensions/src/main/java/io/druid/data/input/parquet/ParquetExtensionsModule.java]: Is it fully compatible? Couldn't somebody depend on the actual name of the module somehow?

File: processing/src/main/java/io/druid/query/IntervalChunkingQueryRunner.java [hunk 2]
```diff
@@ -113,7 +113,8 @@ public IntervalChunkingQueryRunner(
                         ),
                         executor, queryWatcher
                     ).run(
-                        query.withQuerySegmentSpec(new MultipleIntervalSegmentSpec(Arrays.asList(singleInterval))),
+                        query.withQuerySegmentSpec(new MultipleIntervalSegmentSpec(
+                            Collections.singletonList(singleInterval))),
```
Comment (leventov) [inline processing/src/main/java/io/druid/query/IntervalChunkingQueryRunner.java]: Last two )) should be on the next line

File: processing/src/main/java/io/druid/query/groupby/having/GreaterThanHavingSpec.java [hunk 3]
```diff
@@ -71,7 +71,7 @@ public boolean eval(Row row)
   public byte[] getCacheKey()
   {
     final byte[] aggBytes = StringUtils.toUtf8(aggregationName);
-    final byte[] valBytes = Bytes.toArray(Arrays.asList(value));
+    final byte[] valBytes = Bytes.toArray(Collections.singletonList(value));
```
Comment (leventov) [inline processing/src/main/java/io/druid/query/groupby/having/GreaterThanHavingSpec.java]: Could be `new byte[] {value.byteValue()}`

File: processing/src/main/java/io/druid/segment/CompressedVSizeIndexedSupplier.java [hunk 4]
```diff
@@ -138,12 +138,12 @@ public static CompressedVSizeIndexedSupplier fromIterable(
 
 
   @Override
-  public IndexedMultivalue<IndexedInts> get()
+  public IndexedMultiValue<IndexedInts> get()
```
Comment (leventov) [inline processing/src/main/java/io/druid/segment/CompressedVSizeIndexedSupplier.java]: I don't think it should be renamed

File: extensions-contrib/parquet-extensions/src/main/java/io/druid/data/input/parquet/ParquetExtensionsModule.java (hunk 1 above)
Comment (asdf2014) [inline extensions-contrib/parquet-extensions/src/main/java/io/druid/data/input/parquet/ParquetExtensionsModule.java]: I think just we can ensure the name of `SimpleModule` be unique, that will be okay.

File: processing/src/main/java/io/druid/query/IntervalChunkingQueryRunner.java (hunk 2 above)
Comment (asdf2014) [inline processing/src/main/java/io/druid/query/IntervalChunkingQueryRunner.java]: I got it.

File: processing/src/main/java/io/druid/query/groupby/having/GreaterThanHavingSpec.java (hunk 3 above)
Comment (asdf2014) [inline processing/src/main/java/io/druid/query/groupby/having/GreaterThanHavingSpec.java]: Great!

File: processing/src/main/java/io/druid/segment/CompressedVSizeIndexedSupplier.java (hunk 4 above)
Comment (asdf2014) [inline processing/src/main/java/io/druid/segment/CompressedVSizeIndexedSupplier.java]: U r right. I'll roll it back.

**Comment by asdf2014** [comment]
`Timeout Problem` happened again... Somebody could help me to rerun the `Travis` job.

```
java.lang.Exception: test timed out after 60000 milliseconds
	at java.lang.Thread.sleep(Native Method)
	at io.druid.server.coordinator.DruidCoordinatorTest.testCoordinatorRun(DruidCoordinatorTest.java:375)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.FailOnTimeout$StatementThread.run(FailOnTimeout.java:74)
```

**Comment by asdf2014** [comment]
`Timeout Problem` happened again... Somebody could help me to rerun the `Travis` job? :(

```
java.lang.Exception: test timed out after 60000 milliseconds
	at java.lang.Thread.sleep(Native Method)
	at io.druid.server.coordinator.DruidCoordinatorTest.testCoordinatorRun(DruidCoordinatorTest.java:375)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.FailOnTimeout$StatementThread.run(FailOnTimeout.java:74)
```

**Comment by jihoonson** [comment]
LGTM. Thanks @asdf2014!

**Comment by leventov** [comment]
@asdf2014 could you please resolve conflicts? Then we can merge this PR.

**Comment by leventov** [comment]
@asdf2014 your latest commit doesn't have druid-io's master HEAD commit as parent, so Github doesn't hide repetitive changes. Could you please add parent to your latest commit (http://stackoverflow.com/a/4264151/648955 might help) and re-push it?

**Comment by leventov** [comment]
In the future please resolve conflicts via `git merge master`, as described here: https://github.com/druid-io/druid/blob/master/CONTRIBUTING.md#github-workflow

**Comment by asdf2014** [comment]
@leventov Hi, Leventov. I executed the `git rebase --root --onto master --preserve-merges` command on `code_refactoring` branch and get the `Successfully rebased and updated refs/heads/code_refactoring.` message. Is that ok?

**Comment by asdf2014** [comment]
If that is fine.  I would push it with the `git push -u origin code_refactoring --force` command.

**Comment by asdf2014** [comment]
@leventov Already Done.

```bash
git checkout master
git checkout -b r2
git apply ../123.patch
git diff master
git status
git diff --name-status | wc -l
git status
git add ./
git status
git diff master
git commit -m 'Improve `collection` related things that reusing a immutable object instead of creating a new object'
git status
git push origin r2:code_refactoring -f
```

**Comment by asdf2014** [comment]
Why i changed nothing, but i got the `The job exceeded the maximum time limit for jobs, and has been terminated.` message from `travis`? It seems like the time length limitation of the time of build job is 50min in travis.org. (https://docs.travis-ci.com/user/customizing-the-build#Build-Timeouts)

Can anyone help me to restart the build job?

**Comment by gianm** [comment]
@asdf2014 looks like the tests have passed now. thanks for the contribution!

File: extensions-core/avro-extensions/src/main/java/io/druid/data/input/avro/SchemaRepoBasedAvroBytesDecoder.java [hunk 5]
```diff
@@ -80,14 +80,12 @@ public GenericRecord parse(ByteBuffer bytes)
     DatumReader<GenericRecord> reader = new GenericDatumReader<>(schema);
     try (ByteBufferInputStream inputStream = new ByteBufferInputStream(Collections.singletonList(bytes))) {
       return reader.read(null, DecoderFactory.get().binaryDecoder(inputStream, null));
-    }
-    catch (EOFException eof) {
+    } catch (EOFException eof) {
```
Comment (leventov) [inline extensions-core/avro-extensions/src/main/java/io/druid/data/input/avro/SchemaRepoBasedAvroBytesDecoder.java]: Actually this violate Druid code style. Former version of code is correct

**Comment by leventov** [comment]
@asdf2014 irrelevant now but actually what you did is rebase of your entire branch, that is also should be avoided. I asked to rebase only the _latest commit_ in the branch, which you was proposing to merge into druid-io/druid/master, `asdf2014:code_refactoring`. That commit used to be called "Fix conflicts".

**Comment by asdf2014** [comment]
@leventov Sorry violate druid code style... I got it, in fact i also liked squash those commits into a single commit. @gianm You are welcome.

**Comment by leventov** [comment]
@asdf2014 for the future PRs - please _**don't squash anything**_ after you submitted a PR and some time has passed so somebody may has already started looking at your PR (and, moreover, if somebody already left some review comments), because reviewers lose the progress and in some cases need to re-review the whole PR, to be sure that they didn't miss anything.

**Comment by asdf2014** [comment]
Okay, you are right. Those messages of commits should be retained.

**Comment by gianm** [comment]
In addition to the messages, Github also has a nice feature that lets reviewers see what has changed since their last review. It's based on diffing against new commits though, so squashing prevents that feature from working.

FWIW we only very recently changed our guidelines to suggest that people avoid squashing (#4087). Until then we did suggest squashing, but we changed our minds :)

**Comment by asdf2014** [comment]
Oh... Don't know these, i will pay attention in the future.

**Comment by gianm** [comment]
No worries, and thanks again for the patch.

**Comment by asdf2014** [comment]
Alright, you are welcome.

**Pull request merged**